tests/MongoDB.EntityFrameworkCore.FunctionalTests/Mapping/ValueConverterTests.cs

```diff
@@ -796,4 +797,246 @@ public void ObjectId_can_deserialize_and_query_from_string_default(string id)
         var found = db.Entities.First(e => e._id == ObjectId.Parse(id));
         Assert.Equal(id, found._id.ToString());
     }
+
+    [Fact]
+    public void Custom_struct_can_be_used_as_key()
+    {
+        var idValue = HexStruct.Create();
+        const string expectedName = "Any Name Works";
+        var docs = database.CreateCollection<IdIsStringWithName>();
+        docs.InsertOne(new IdIsStringWithName {_id = idValue.ToString(), name = expectedName});
+
+        var collection = database.GetCollection<IdGenericEntity<HexStruct>>(docs.CollectionNamespace);
+        using var db = SingleEntityDbContext.Create(collection, mb =>
+        {
+            mb.Entity<IdGenericEntity<HexStruct>>(e =>
+            {
+                e.HasKey(f => f.Id);
+                e.Property(f => f.Id).HasElementName("_id").HasConversion(k => k.ToString(), s => new HexStruct(s));
+                e.Property(f => f.Name).HasElementName("name");
+            });
+        });
+
+        var found = db.Entities.First(e => e.Id.Equals(idValue));
+        Assert.Equal(idValue, found.Id);
+        Assert.Equal(expectedName, found.Name);
+
+        found.Name = "Test";
```

Comment:
What is this doing?

Reply:
It's just making sure it can save the property as well (db.SaveChanges does that, EF tracks property changes)